### PR TITLE
Add Annika

### PR DESCRIPTION
**Pull request by AnnikaCodes**

### CODE REVIEW

File: data/mods/ssb/abilities.ts [hunk 1]
```diff
@@ -37,7 +37,15 @@ export function changeSet(context: Battle, pokemon: Pokemon, newSet: SSBSet) {
...
 	pokemon.setItem(item);
+	changeMoves(context, pokemon, newSet);
+}
 
+/**
+ * Assigns new moves to a Pokemon
+ * @param pokemon The Pokemon whose moveset is to be modified
+ * @param newSet The set whose moves should be assigned
+ */
+export function changeMoves(context: Battle, pokemon: Pokemon, newSet: SSBSet) {
```
Comment (ghost) [inline data/mods/ssb/abilities.ts:48]: Im gonna be honest, this looks incredibly niche as youre the only one who is using this

Comment (AnnikaCodes) [reply data/mods/ssb/abilities.ts:48]: I didn't want to duplicate code between `changeSet` and Refactor, though.

Comment (AnnikaCodes) [reply data/mods/ssb/abilities.ts:48]: (`changeSet` calls `changeMoves`)

File: data/mods/ssb/moves.ts [hunk 2]
```diff
@@ -265,6 +265,27 @@ export const Moves: {[k: string]: ModdedMoveData} = {
...
+		desc: "Raises Special Attack by 1 stage and regenerates user and foe movesets, restoring PP.",
+		shortDesc: "Regenerates user, foe movesets; +1 Sp. Atk.",
+		name: "Refactor",
+		pp: 5,
+		flags: {authentic: 1},
+		priority: -7,
+		onHit(target, source, move) {
+			for (const pokemon of this.getAllActive()) {
+				changeMoves(this, pokemon, ssbSets[pokemon.name]);
+				this.add('-message', `Annika refactored ${pokemon === target ? `herself` : `the opposing ${pokemon.name}`}!`);
```
Comment (ghost) [inline data/mods/ssb/moves.ts]:
```suggestion
				this.add('-message', `${source.name} refactored ${pokemon === target ? `themself` : `the opposing ${pokemon.name}`}!`);
```

Comment (ghost) [reply data/mods/ssb/moves.ts]: Note: Its not just you who can use this move

File: data/mods/ssb/abilities.ts (hunk 1 above)
Comment (ghost) [reply data/mods/ssb/abilities.ts:48]: Gotcha

File: data/mods/ssb/moves.ts (hunk 2 above)
Comment (AnnikaCodes) [reply data/mods/ssb/moves.ts]: Shouldn't it also be `${target.name} refactored themselves` then?

Comment (ghost) [reply data/mods/ssb/moves.ts]: Im not good at words. So probably ye, ill update it

**Pull request merged**